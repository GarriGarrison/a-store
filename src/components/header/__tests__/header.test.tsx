import { render, screen, fireEvent } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Header } from '..'


// const handleMenuOpen = jest.fn()

describe('Тесты компонента Header', () => {
  test('рендер изображения - логотипа', () => {
    render(
      <Router>
        <Header />
      </Router>
    )

    const element = screen.getByTestId('logo')
    expect(element).toBeInTheDocument()
  })


  test('рендер изображения - меню', () => {
    render(
      <Router>
        <Header />
      </Router>
    )

    const element = screen.getByTestId('menu')
    expect(element).toBeInTheDocument()
  })


  test('рендер текста - меню', () => {
    render(
      <Router>
        <Header />
      </Router>
    )

    const element = screen.getByText('МЕНЮ')
    expect(element).toBeInTheDocument()
  })


  test('снепшот компонента', () => {
    const { baseElement } = render(
      <Router>
        <Header />
      </Router>
    )

    expect(baseElement).toMatchSnapshot()
  })
})
