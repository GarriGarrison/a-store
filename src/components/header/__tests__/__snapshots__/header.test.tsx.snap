// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты компонента Header снепшот компонента 1`] = `
<body>
  <div>
    <header
      class="container"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          class="logo"
          data-testid="logo"
          src="logo.png"
        />
      </a>
      <nav>
        <div
          class="menu"
        >
          <img
            alt="menu"
            class="image"
            data-testid="menu"
            src="menu.svg"
          />
          <span>
            МЕНЮ
          </span>
        </div>
      </nav>
    </header>
  </div>
</body>
`;
